### PR TITLE
Adapt Godwoken v1.1

### PR DESCRIPTION
**Pull request by Flouse**
## Godwoken v1.1 Changes
- https://github.com/nervosnetwork/godwoken/blob/e7be07535e24e83d2dc546bd659d1dc90e7a2850/docs/CHANGES.md

## Related PR
- https://github.com/nervosnetwork/godwoken/pull/649
- https://github.com/nervosnetwork/godwoken-web3/pull/282

### CODE REVIEW

File: docker/layer2/entrypoint.sh [hunk 1]
```diff
@@ -128,7 +128,8 @@ function generate-godwoken-config() {
         --rollup-config $CONFIG_DIR/rollup-config.json \
         --privkey-path $ACCOUNTS_DIR/godwoken-block-producer.key \
         -o $CONFIG_DIR/godwoken-config.toml \
-        --rpc-server-url 0.0.0.0:8119
+        --rpc-server-url 0.0.0.0:8119 \
+        --block-producer-address 0x2e9df163055245bfadd35e3a1f05f06096447c85
```
Comment (keroro520) [inline docker/layer2/entrypoint.sh:132]: Please add a code comment to describe that the private key of `0x2e9df163055245bfadd35e3a1f05f06096447c85` is `accounts/godwoken-block-producer.key`

File: docker/layer2/entrypoint.sh [hunk 2]
```diff
@@ -138,7 +139,7 @@ function generate-godwoken-config() {
         sed -i 's#^path = .*$#path = '"'$STORE_PATH'"'#' $CONFIG_DIR/godwoken-config.toml
     fi
     sed -i 's#enable_methods = \[\]#err_receipt_ws_listen = '"'0.0.0.0:8120'"'#' $CONFIG_DIR/godwoken-config.toml
-    config-godwoken-eoa-register
+    # config-godwoken-eoa-register
```
Comment (keroro520) [inline docker/layer2/entrypoint.sh:142]: Remove the useless code

File: docker/layer2/entrypoint.sh [hunk 3]
```diff
@@ -222,33 +197,31 @@ ROLLUP_TYPE_HASH=$(jq -r '.rollup_type_hash' $CONFIG_DIR/rollup-genesis-deployme
...
 
 # When requests "executeTransaction" RPC interface, the RawL2Transaction's
 # signature can be omit. Therefore we fill the RawL2Transaction.from_id
 # with this DEFAULT_FROM_ID (corresponding to DEFAULT_FROM_ADDRESS).
-DEFAULT_FROM_ID=2
+DEFAULT_FROM_ID=3
 DEFAULT_FROM_ADDRESS=0x2e9df163055245bfadd35e3a1f05f06096447c85
 
-ETH_ADDRESS_REGISTRY_ACCOUNT_ID=3
+ETH_ADDRESS_REGISTRY_ACCOUNT_ID=2
```
Comment (keroro520) [inline docker/layer2/entrypoint.sh:224]: A link to the source definition (Godwoken source code) would be helpful for reader's understanding.

File: docker/layer2/entrypoint.sh [hunk 4]
```diff
@@ -222,33 +197,31 @@ ROLLUP_TYPE_HASH=$(jq -r '.rollup_type_hash' $CONFIG_DIR/rollup-genesis-deployme
...
+# see: https://github.com/nervosnetwork/godwoken/blob/develop/docs/life_of_a_polyjuice_transaction.md#root-account--deployment
 CREATOR_ACCOUNT_ID=$creator_account_id
-COMPATIBLE_CHAIN_ID=$COMPATIBLE_CHAIN_ID
-CHAIN_ID=$((($COMPATIBLE_CHAIN_ID << 32) + $creator_account_id))
 
 # When requests "executeTransaction" RPC interface, the RawL2Transaction's
 # signature can be omit. Therefore we fill the RawL2Transaction.from_id
 # with this DEFAULT_FROM_ID (corresponding to DEFAULT_FROM_ADDRESS).
-DEFAULT_FROM_ID=2
+DEFAULT_FROM_ID=3
```
Comment (keroro520) [inline docker/layer2/entrypoint.sh:221]: What does `3` indicate?

A link to the source definition (Godwoken source code) would be helpful for reader's understanding.

File: docker/layer2/entrypoint.sh [hunk 5]
```diff
@@ -315,7 +290,9 @@ function main() {
     start-godwoken-at-background
 
     # Should make sure that the Polyjuice root account was created and the layer2 block was synced
-    create-polyjuice-root-account
+    create-polyjuice-root-account \
+    || echo "Error:Failed to create-polyjuice-root-account"
```
Comment (keroro520) [inline docker/layer2/entrypoint.sh:294]: This script sets `set -o errexit` at the beginning.

Do you want the script to keep running even if `create-polyjuice-root-account` fails?

**Review by keroro520 — CHANGES_REQUESTED** [review]

File: docker/layer2/entrypoint.sh (hunk 4 above)
Comment (Flouse) [reply docker/layer2/entrypoint.sh:221]:
```sh
# When requesting to "executeTransaction" RPC interface, the RawL2Transaction's
# signature can be omit. Therefore we fill the RawL2Transaction.from_id
# with this DEFAULT_FROM_ID (corresponding to DEFAULT_FROM_ADDRESS).
#
# Usually, ID(3) indicates the first deposited Godwoken account.
DEFAULT_FROM_ID=3
DEFAULT_FROM_ADDRESS=0x2e9df163055245bfadd35e3a1f05f06096447c85
```

File: docker/layer2/entrypoint.sh (hunk 3 above)
Comment (Flouse) [reply docker/layer2/entrypoint.sh:224]:
```
# The builtin ETH_ADDRESS_REGISTRY_ACCOUNT_ID is 2.
# see: https://github.com/nervosnetwork/godwoken/blob/6180f04/crates/common/src/builtins.rs#L5
ETH_ADDRESS_REGISTRY_ACCOUNT_ID=2
```

File: docker/layer2/entrypoint.sh (hunk 1 above)
Comment (Flouse) [reply docker/layer2/entrypoint.sh:132]:
```sh
# Node: 0x2e9df163055245bfadd35e3a1f05f06096447c85 is the eth_address of
    # `godwoken-block-producer.key`
    RUST_BACKTRACE=full gw-tools generate-config \
        --ckb-rpc http://ckb:8114 \
        --ckb-indexer-rpc http://ckb-indexer:8116 \
        -c $CONFIG_DIR/scripts-config.json \
        --scripts-deployment-path $CONFIG_DIR/scripts-deployment.json \
        --omni-lock-config-path $CONFIG_DIR/scripts-deployment.json \
        -g $CONFIG_DIR/rollup-genesis-deployment.json \
        --rollup-config $CONFIG_DIR/rollup-config.json \
        -o $CONFIG_DIR/godwoken-config.toml \
        --rpc-server-url 0.0.0.0:8119 \
        --privkey-path $ACCOUNTS_DIR/godwoken-block-producer.key \
        --block-producer-address 0x2e9df163055245bfadd35e3a1f05f06096447c85
```

**Review by keroro520 — APPROVED** [review]

**Pull request merged**